packages/enzyme/src/selectors.js

```diff
@@ -55,6 +64,80 @@ function safelyGenerateTokens(selector) {
   }
 }
 
+function matchAttributeSelector(node, token) {
+  const { operator, value, name } = token;
+  const nodeProps = propsOfNode(node);
+  const descriptor = Object.getOwnPropertyDescriptor(nodeProps, name);
+  if (descriptor && descriptor.get) {
+    return false;
+  }
+  const nodePropValue = nodeProps[name];
+  if (typeof nodePropValue === 'undefined') {
+    return false;
+  }
+  if (token.type === ATTRIBUTE_PRESENCE) {
+    return has(nodeProps, token.name);
+  }
+  // Only the exact value operator ("=") can match non-strings
+  if (typeof nodePropValue !== 'string' || typeof value !== 'string') {
+    if (operator !== EXACT_ATTRIBUTE_OPERATOR) {
+      return false;
+    }
+  }
+  switch (operator) {
+    /**
+     * Represents an element with the att attribute whose value is exactly "val".
+     * @example
+     * [attr="val"] matches attr="val"
+     */
+    case EXACT_ATTRIBUTE_OPERATOR:
+      return is(nodePropValue, value);
+    /**
+     * Represents an element with the att attribute whose value is a whitespace-separated
+     * list of words, one of which is exactly 
+     * @example
+     *  [rel~="copyright"] matches rel="copyright other"
+     */
+    case WHITELIST_ATTRIBUTE_OPERATOR:
+      return nodePropValue.split(' ').indexOf(value) !== -1;
```

Comment:
This would throw an error if `nodePropValue` isn't a string - is it safe to assume that if someone's using `~=`, they're testing a string? Or would it be safer to return false if not?

Reply:
There's [an earlier check](https://github.com/airbnb/enzyme/pull/1157/files#diff-ee4c070528cfb6bb4936ac945d73692bR82) that tests for non-string values, so we can safely assume that `nodePropValue` and `value` are both strings in this case.

> is it safe to assume that if someone's using ~=, they're testing a string?

Yupp, that's exactly the assumption! These operators don't make a whole lot of sense with other values like numbers or objects.

Reply:
I must've missed that. It makes a lot more sense now, thanks for explaining!